localstack/services/stepfunctions/asl/component/state/state_execution/state_task/service/state_task_service_ecs.py

```diff
@@ -0,0 +1,127 @@
+    StateTaskServiceCallback,
+)
+from localstack.services.stepfunctions.asl.eval.environment import Environment
+from localstack.services.stepfunctions.asl.utils.boto_client import boto_client_for
+
+
+class StateTaskServiceEcs(StateTaskServiceCallback):
+    _SUPPORTED_API_PARAM_BINDINGS: Final[dict[str, set[str]]] = {
+        "runtask": {
+            "Cluster",
+            "Group",
+            "LaunchType",
+            "NetworkConfiguration",
+            "Overrides",
+            "PlacementConstraints",
+            "PlacementStrategy",
+            "PlatformVersion",
+            "PropagateTags",
+            "TaskDefinition",
+            "EnableExecuteCommand",
+        }
+    }
+
+    _STARTED_BY_PARAMETER_RAW_KEY: Final[str] = "StartedBy"
+    _STARTED_BY_PARAMETER_VALUE: Final[str] = "AWS Step Functions"
+
+    def _get_supported_parameters(self) -> Optional[set[str]]:
+        return self._SUPPORTED_API_PARAM_BINDINGS.get(self.resource.api_action.lower())
+
+    def _before_eval_execution(
+        self, env: Environment, resource_runtime_part: ResourceRuntimePart, raw_parameters: dict
+    ) -> None:
+        if self.resource.condition == ResourceCondition.Sync:
+            raw_parameters[self._STARTED_BY_PARAMETER_RAW_KEY] = self._STARTED_BY_PARAMETER_VALUE
+        super()._before_eval_execution(
+            env=env, resource_runtime_part=resource_runtime_part, raw_parameters=raw_parameters
+        )
+
+    def _eval_service_task(
+        self,
+        env: Environment,
+        resource_runtime_part: ResourceRuntimePart,
+        normalised_parameters: dict,
+    ):
+        service_name = self._get_boto_service_name()
+        api_action = self._get_boto_service_action()
+        ecs_client = boto_client_for(
+            region=resource_runtime_part.region,
+            account=resource_runtime_part.account,
+            service=service_name,
+        )
+        response = getattr(ecs_client, api_action)(**normalised_parameters)
+        response.pop("ResponseMetadata", None)
+
+        # AWS outputs the description of the task, not the output of run_task.
+        match self._get_boto_service_action():
+            case "run_task":
+                self._normalise_response(response=response, service_action_name="run_task")
+                cluster_arn: str = response["Tasks"][0]["ClusterArn"]
+                task_arn: str = response["Tasks"][0]["TaskArn"]
+                describe_tasks_output = ecs_client.describe_tasks(
+                    cluster=cluster_arn, tasks=[task_arn]
+                )
+                describe_tasks_output.pop("ResponseMetadata", None)
+                self._normalise_response(
+                    response=describe_tasks_output, service_action_name="describe_tasks"
+                )
+                env.stack.append(describe_tasks_output)
+                return
+
+        env.stack.append(response)
```

Comment:
do we even want to have this as a fallback, given that this shouldn't be valid?

Reply:
My thought was it could ease bug reports if we see the invalid valued being passed on, but yes technically it can be removed.

Reply:
My argument against this would be that this hides an issue that the user needs to fix in his state machine and won't deploy like this against AWS as well, so I don't see a reason to hide it. It will also make potential bug reports harder to debug.